### PR TITLE
Added Link Type check and simplified floating link plugin

### PR DESCRIPTION
**Pull request by HarrySIV**
Extension of #5551 
@zurfyx is this closer to what you were looking for? I certainly made it simple for the product side, but I'm not sure if there's a more optimal solution, or if this is overkill. 
Also I noticed $getAncestor and $findMatchingParent are similar but I'm not sure if there's a distinct difference I'm not seeing.

### CODE REVIEW

File: packages/lexical-link/src/index.ts [hunk 1]
```diff
@@ -535,3 +540,69 @@ function $getAncestor<NodeType extends LexicalNode = LexicalNode>(
...
+  const firstPoint = isBefore ? anchor : focus;
+  const lastPoint = isBefore ? focus : anchor;
+  const firstNode = firstPoint.getNode();
+  const lastNode = lastPoint.getNode();
+  const firstLinkNode = getParentLink(firstNode);
+  const lastLinkNode = getParentLink(lastNode);
+  if (
+    !firstLinkNode ||
+    !lastLinkNode ||
+    (firstLinkNode && !firstLinkNode.is(lastLinkNode))
```
Comment (zurfyx) [inline packages/lexical-link/src/index.ts:565]: Doesn't this lead to the same problem? The first node can be a link while the last node points to a LineBreakNode

File: packages/lexical-link/src/index.ts [hunk 2]
```diff
@@ -535,3 +540,69 @@ function $getAncestor<NodeType extends LexicalNode = LexicalNode>(
...
+        } else {
+          continue;
+        }
+      } else {
+        matchedParent = node;
+        continue;
+      }
+    }
+    const parent = getParentLink(node);
+    if (!parent) {
```
Comment (zurfyx) [inline packages/lexical-link/src/index.ts:588]:
```suggestion
    if (parent === null) {
```

File: packages/lexical-link/src/index.ts [hunk 3]
```diff
@@ -535,3 +540,69 @@ function $getAncestor<NodeType extends LexicalNode = LexicalNode>(
...
+      return false;
+    } else if (!matchedParent) {
```
Comment (zurfyx) [inline packages/lexical-link/src/index.ts:590]:
```suggestion
    } else if (matchedParent === null) {
```

File: packages/lexical-link/src/index.ts [hunk 4]
```diff
@@ -535,3 +540,69 @@ function $getAncestor<NodeType extends LexicalNode = LexicalNode>(
...
+
+/**
+ * Checks selection to ensure only a single LinkNode or AutoLinkNode is being selected,
+ * selecting any Node that doesn't have a some Link as a parent will return false.
+ * @param selection - The current range selection.
+ * @returns - true if selection only contains a single LinkNode or AutoLinkNode, false otherwise.
+ */
+export function $onlyIncludesAParentLink(
+  selection: RangeSelection,
+): boolean | null {
```
Comment (zurfyx) [inline packages/lexical-link/src/index.ts:552]:
```suggestion
export function $getSelectedLinkNode(
  selection: RangeSelection,
): null | LinkNode {
```

File: packages/lexical-link/src/index.ts [hunk 5]
```diff
@@ -535,3 +540,69 @@ function $getAncestor<NodeType extends LexicalNode = LexicalNode>(
...
+    const parent = getParentLink(node);
+    if (!parent) {
+      return false;
+    } else if (!matchedParent) {
+      matchedParent = parent;
+    } else if (!parent.is(matchedParent)) {
+      return false;
+    }
+  }
+  return true;
```
Comment (zurfyx) [inline packages/lexical-link/src/index.ts:596]: As suggested above, let's change this for the actual matched LinkNode

**Review by zurfyx — COMMENTED** [review]
Added some minor comments but for the most part LGTM, thank you!

**Comment by HarrySIV** [comment]
So I ended up having to delve further into `getNodesBetween` to get around Lexical being incapable of selecting LineBreakNodes. The big issue I couldn't solve is when you select a LineBreakNode on the beginning or end of selection, lexical selects the entire paragraph node anyway. 

So there's a weird case where you have a linebreak and a link, and if you select just the linebreak only, lexical thinks you're selecting the whole paragraph and, therefore, the link. I'm not sure how to work around this.

**Comment by etrepum** [comment]
The selection issues should be solved by #6883, but this PR is quite stale and has conflicts. Please re-open if you intend to update it to work with the current version.

**Pull request closed without merging**